root/etc/e-smith/templates/etc/ssh/sshd_config/70Restricted2Sftp

```diff
@@ -0,0 +1,23 @@
+{
+    # restrict groups to sftp, ssh access is refused
+    my $domain = $DomainName || die ('Cannot retrieve DomainName');
+    my @SftpRestricted = ();
+
+    foreach ( split(',',$sshd{'AllowGroups'} || '')) {
+        my ($group, $sftp) = split(':', $_);
+        if(!$group || !$sftp) {
+            next;
+        }
+        $group .= "\@$domain" if ($group !~ '@');
+        push @SftpRestricted, $group if ($sftp eq 'sftp');
+    }
+
+    if (@SftpRestricted) {
+        my $matchGroup = join (",", map { $_ } @SftpRestricted);
```

Comment:
The `map()` is useless here. Please, see the next comment.

Reply:
we need a comma separated list see the explanation below